File: src/features/symbols/types/symbol.ts
export type SymbolMarket = 'crypto' | 'forex' | 'metals' | 'indices' | 'stocks'
export type AssetClass =
  | 'Forex'
  | 'Cryptocurrencies'
  | 'Metals'
  | 'Indices'
  | 'Stocks'
  | 'Shares'
  | 'ETFs'
  | 'Energies'
  | 'Commodities'

export interface AdminSymbol {
  id: string
  symbolCode: string
  /** Alias for symbolCode - for compatibility */
  code?: string
  /** Display name - for compatibility */
  name?: string
  /** Market/category - for compatibility */
  market?: string
  providerSymbol: string | null
  assetClass: AssetClass | null
  baseCurrency: string
  quoteCurrency: string
  pricePrecision: number
  volumePrecision: number
  contractSize: string
  tickSize?: number | null
  lotMin?: number | null
  lotMax?: number | null
  defaultPipPosition?: number | null
  pipPositionMin?: number | null
  pipPositionMax?: number | null
  isEnabled: boolean
  tradingEnabled: boolean
  leverageProfileId: string | null
  leverageProfileName: string | null
  createdAt: string
  updatedAt: string
}

export interface CreateSymbolPayload {
  symbol_code: string
  provider_symbol: string
  asset_class: AssetClass
  base_currency: string
  quote_currency: string
  price_precision: number
  volume_precision: number
  contract_size: string
  tick_size?: string | null
  lot_min?: string | null
  lot_max?: string | null
  default_pip_position?: string | null
  pip_position_min?: string | null
  pip_position_max?: string | null
  leverage_profile_id?: string | null
}

export interface UpdateSymbolPayload {
  symbol_code: string
  provider_symbol: string
  asset_class: AssetClass
  base_currency: string
  quote_currency: string
  price_precision: number
  volume_precision: number
  contract_size: string
  tick_size?: string | null
  lot_min?: string | null
  lot_max?: string | null
  default_pip_position?: string | null
  pip_position_min?: string | null
  pip_position_max?: string | null
  is_enabled: boolean
  trading_enabled: boolean
  leverage_profile_id?: string | null
}

export interface ListSymbolsParams {
  search?: string
  asset_class?: string
  is_enabled?: string
  page?: number
  page_size?: number
  sort?: string
}

export interface ListSymbolsResponse {
  items: AdminSymbol[]
  page: number
  page_size: number
  total: number
}

export interface GroupMarkup {
  groupId: string
  groupName: string
  markupType: 'percent'
  markupValue: number
  enabled: boolean
}

File: backend/auth-service/src/models/symbol.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Symbol {
    pub id: Uuid,
    #[sqlx(rename = "symbol_code")]
    pub symbol_code: String,
    pub provider_symbol: Option<String>,
    pub asset_class: Option<String>,
    pub base_currency: String,
    pub quote_currency: String,
    pub price_precision: i32,
    pub volume_precision: i32,
    pub contract_size: String,
    #[sqlx(rename = "tick_size")]
    pub tick_size: Option<rust_decimal::Decimal>,
    #[sqlx(rename = "lot_min")]
    pub lot_min: Option<rust_decimal::Decimal>,
    #[sqlx(rename = "lot_max")]
    pub lot_max: Option<rust_decimal::Decimal>,
    #[sqlx(rename = "default_pip_position")]
    pub default_pip_position: Option<rust_decimal::Decimal>,
    #[sqlx(rename = "pip_position_min")]
    pub pip_position_min: Option<rust_decimal::Decimal>,
    #[sqlx(rename = "pip_position_max")]
    pub pip_position_max: Option<rust_decimal::Decimal>,
    pub is_enabled: bool,
    pub trading_enabled: bool,
    pub leverage_profile_id: Option<Uuid>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct SymbolWithProfile {
    pub id: Uuid,
    #[sqlx(rename = "symbol_code")]
    pub symbol_code: String,
    pub provider_symbol: Option<String>,
    pub asset_class: Option<String>,
    pub base_currency: String,
    pub quote_currency: String,
    pub price_precision: i32,
    pub volume_precision: i32,
    #[sqlx(rename = "contract_size")]
    pub contract_size: String,
    #[sqlx(rename = "tick_size")]
    pub tick_size: Option<rust_decimal::Decimal>,
    #[sqlx(rename = "lot_min")]
    pub lot_min: Option<rust_decimal::Decimal>,
    #[sqlx(rename = "lot_max")]
    pub lot_max: Option<rust_decimal::Decimal>,
    #[sqlx(rename = "default_pip_position")]
    pub default_pip_position: Option<rust_decimal::Decimal>,
    #[sqlx(rename = "pip_position_min")]
    pub pip_position_min: Option<rust_decimal::Decimal>,
    #[sqlx(rename = "pip_position_max")]
    pub pip_position_max: Option<rust_decimal::Decimal>,
    pub is_enabled: bool,
    pub trading_enabled: bool,
    pub leverage_profile_id: Option<Uuid>,
    pub leverage_profile_name: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

